synapse/cores/common.py

```diff
@@ -2703,7 +2703,10 @@ def _runToAdd(self, fulls):
                 if self.isRuntForm(stype):
                     continue
 
-                if self.isTufoForm(stype):
+                if not self.isTufoForm(stype):
+                    continue
+
+                if self.isAutoAddSafe(stype):
```

Comment:
In the given example, `inet:web:post` will come in fulls as the guid, so there is no way to add the props.

Reply:
Preventing the nodes from being formed incorrectly is definitely better, but I'd still like to work together to have the nodes form correctly.

Reply:
Maybe instead of checking during `_runToAdd`, we check during or after the call to `_normTufoProps` and if the type isn't save to use that way, we pass along the valus that we originally got for it?

Reply:
@vertexmc I think xref is probably a worst case scenario for this type of problem, and probably the best case to decide "how to do things".  I did a quick tweak to xref's to also kick out subs for the primary property node when it does norming.  This resulted in make the following props as we move through formTufoByProp:

```
forming inet:web:postref = [(('vertex.link/user', 'mypost 0.0.0.0'), ('inet:ipv4', 0))]
inprops: {}
updated props: {'post:acct:user': 'user', 'post:acct:site': 'vertex.link', 'post:acct': 'vertex.link/user', 'xref:prop': 'inet:ipv4', 'xref': 'inet:ipv4=0.0.0.0', 'post:text': 'mypost 0.0.0.0', 'post:acct:site:host': 'vertex', 'xref:intval': 0, 'post': '68bc4607f0518963165536921d6e86fa', 'post:acct:site:domain': 'link'}
fulls: {'inet:web:postref:xref:intval': 0, 'inet:web:postref:xref': 'inet:ipv4=0.0.0.0', 'inet:web:postref:xref:prop': 'inet:ipv4', 'inet:web:postref:post': '68bc4607f0518963165536921d6e86fa'}
```

We'd have to identify the hierarchy during _normTufoProps and fire autoadds with props ganked from the namespace in order to correctly form the nodes made with these types.

Reply:
I think this comes down to the decision "we try our best to make the thingies" or "we safely fail to make certain times of thingies to avoid data loss"

Reply:
I also think that if we do try to make the other nodes, we need to avoid needing to add extra props to the xref nodes.

Reply:
If we merged the PR as-is, I could just make sure to always form the postref and the post, but that seems like a lot of overhead, not just in terms of requests, but also in weirdness that I need to know about as a dev.